pkg/controller/migration/convert/core.go

```diff
@@ -9,6 +13,73 @@ func handleCore(c *components, install *Installation) error {
 		return ErrIncompatibleCluster{"only DATASTORE_TYPE=kubernetes is supported at this time"}
 	}
 
+	// node resource limits
+	node := getContainer(c.node.Spec.Template.Spec, "calico-node")
+	if len(node.Resources.Limits) > 0 || len(node.Resources.Requests) > 0 {
+		install.Spec.ComponentResources = append(install.Spec.ComponentResources, &operatorv1.ComponentResource{
+			ComponentName:        operatorv1.ComponentNameNode,
+			ResourceRequirements: node.Resources.DeepCopy(),
+		})
+	}
+
+	// kube-controllers
+	kubeControllers := getContainer(c.kubeControllers.Spec.Template.Spec, "calico-kube-controllers")
+	if len(kubeControllers.Resources.Limits) > 0 || len(kubeControllers.Resources.Requests) > 0 {
+		install.Spec.ComponentResources = append(install.Spec.ComponentResources, &operatorv1.ComponentResource{
+			ComponentName:        operatorv1.ComponentNameKubeControllers,
+			ResourceRequirements: kubeControllers.Resources.DeepCopy(),
+		})
+	}
+
+	// typha resource limits. typha is optional so check for nil first
+	typha := getContainer(c.typha.Spec.Template.Spec, "calico-typha")
+	if typha != nil && (len(typha.Resources.Limits) > 0 || len(typha.Resources.Requests) > 0) {
+		install.Spec.ComponentResources = append(install.Spec.ComponentResources, &operatorv1.ComponentResource{
+			ComponentName:        operatorv1.ComponentNameTypha,
+			ResourceRequirements: typha.Resources.DeepCopy(),
+		})
+	}
+
+	// kube-controllers nodeSelector
+	install.Spec.ControlPlaneNodeSelector = c.kubeControllers.Spec.Template.Spec.NodeSelector
+
+	// node update-strategy
+	install.Spec.NodeUpdateStrategy = c.node.Spec.UpdateStrategy
+
+	// node volumes
+	for _, vol := range c.node.Spec.Template.Spec.Volumes {
+		switch vol.Name {
+		case "flexvol-driver-host":
+			// prefer user-defined flexvolpath over detected value
+			if install.Spec.FlexVolumePath == "" {
+				if vol.HostPath == nil {
+					return ErrIncompatibleCluster{"volume 'flexvol-driver-host' must be a HostPath"}
+				}
+				install.Spec.FlexVolumePath = vol.HostPath.Path
+			}
+		}
+	}
+
+	// alp
+	vol := getVolume(c.node.Spec.Template.Spec, "flexvol-driver-host")
+	if vol != nil {
+		// prefer user-defined flexvolpath over detected value
+		if install.Spec.FlexVolumePath == "" {
+			if vol.HostPath == nil {
+				return ErrIncompatibleCluster{"volume 'flexvol-driver-host' must be a HostPath"}
+			}
+			if fv := getContainer(c.node.Spec.Template.Spec, "flexvol-driver"); fv == nil {
+				return ErrIncompatibleCluster{"detected 'flexvol-driver-host' volume but no 'flexvol-driver' init container"}
+			}
+			install.Spec.FlexVolumePath = vol.HostPath.Path
+		}
+	} else {
+		// verify that no flexvol container is set
```

Comment:
I think here we should be setting `install.Spec.FlexVolumePath = None`. That's how we disable it with the operator installation.

Reply:
I think this branch should end with setting `install.Spec.FlexVolumePath = None`. It is not sufficient to leave it unset because when our defaulting logic runs we would set a default path.